test/gcs.js

```diff
@@ -0,0 +1,169 @@
+/* global describe, it */
+const assert = require('assert');
+const request = require('request');
+
+describe('UploadFS GCS', function () {
+  this.timeout(20000);
+  const uploadfs = require('../uploadfs.js')();
+  const fs = require('fs');
+  const async = require('async');
+  const tempPath = '../temp';
+  const dstPath = '/one/two/three/test.txt';
+  const imageSizes = [
+    {
+      name: 'small',
+      width: 320,
+      height: 320
+    },
+    {
+      name: 'medium',
+      width: 640,
+      height: 640
+    },
+    {
+      name: 'large',
+      width: 1140,
+      height: 1140
+    }
+  ];
+
+  let gcsOptions = require('../gcsTestOptions.js');
+
+  gcsOptions.imageSizes = imageSizes;
+  gcsOptions.tempPath = tempPath;
+  gcsOptions.params = {
+    Bucket: gcsOptions.bucket,
+    ACL: 'public-read'
```

Comment:
Is this a requirement that needs to be in the documentation? What does it mean to set the ACL at this level rather than via makePublic one file at a time - is that needed?

Reply:
Does this mean the public can enumerate everything in the bucket?

Reply:
TL;DR, yes to both questions. 

I took the s3 tests "at their word" by copying them and making my version of the backend obey the same assertions. One of them was to assert that putting a resource into the remote bucket made it readable by an ordinary http client with no special access control. The GCS bucket I used for my testing was not public, only individual resources were made that way, so they should not be "enumeratable", but if you know the resource is there, it should be readable by anyone.

Let me know if you think this is reasonable.

Reply:
On second thought I think this part was slavishly copied from the s3 test. It has no effect on the ACLs on GCS, which is why I had to make it public when you `copyIn` elsewhere. I'll remove this. It's a canard.